### PR TITLE
dotcopy: init at 0.2.14

### PR DESCRIPTION
**Pull request by FireSquid6**
## Description of changes

This is my first pull request to a large codebase. Please let me know if I am doing anything horribly wrong. I'm trying my best.

Dotcopy is a Linux dotfiles manager that allows you to make templates for your dotfiles to easily share them across multiple machines that require slight differences. 

See [the github](https://github.com/firesquid6/dotcopy)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (I am like 90% sure a raspberry pi counts for this)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) (I don't think this is necessary?)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/do/dotcopy/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, buildGoModule
+, fetchzip
+}:
+
+buildGoModule rec {
+  pname = "dotcopy";
+  version = "0.2.14";
+
+  src = fetchzip {
```
Comment (dotlambda) [inline pkgs/by-name/do/dotcopy/package.nix:10]:
```suggestion
  src = fetchFromGitHub {
```

File: pkgs/by-name/do/dotcopy/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    url = "https://github.com/FireSquid6/dotcopy/archive/refs/tags/v${version}.zip";
+    hash = "sha256-oVMsIZUJ7xOBwSlJF+RUIYG0dPMTZ3ftDd9cpRytl7w=";
+  };
+
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/by-name/do/dotcopy/package.nix:15]: Isn't that the default? If not add a comment with the reason.

File: maintainers/maintainer-list.nix
Comment (dotlambda) [inline maintainers/maintainer-list.nix:1]: Only modify the lines for your entry.

File: pkgs/by-name/do/dotcopy/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  vendorSha256 = null;
+
+  subPackages = [ "." ];
+
+  meta = with lib; {
+    description = "A linux dotfile manager";
+    homepage = "https://dotcopy.firesquid.co";
```
Comment (dotlambda) [inline pkgs/by-name/do/dotcopy/package.nix:22]: 404

File: pkgs/by-name/do/dotcopy/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    license = licenses.gpl3;
```
Comment (dotlambda) [inline pkgs/by-name/do/dotcopy/package.nix:23]: gpl3Plus or gpl3Only?

**Review by dotlambda — COMMENTED** [review]
Please squash into two commits:

* `maintainers: add firesquid6`
* `dotcopy: init at 0.2.14`

File: pkgs/by-name/do/dotcopy/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "dotcopy";
+  version = "0.2.14";
+
+  src = fetchzip {
+    url = "https://github.com/FireSquid6/dotcopy/archive/refs/tags/v${version}.zip";
+    hash = "sha256-oVMsIZUJ7xOBwSlJF+RUIYG0dPMTZ3ftDd9cpRytl7w=";
+  };
+
+  doCheck = false;
+  vendorSha256 = null;
```
Comment (dotlambda) [inline pkgs/by-name/do/dotcopy/package.nix:16]:
```suggestion
  vendorHash = null;
```

File: pkgs/by-name/do/dotcopy/package.nix (hunk 1 above)
Comment (FireSquid6) [reply pkgs/by-name/do/dotcopy/package.nix:10]: I was originally using fetchFromGitHub and it wasn't working. Details of my suffering are here: [https://discourse.nixos.org/t/buildgomodule-fails-due-to-invalid-go-version/40823](https://discourse.nixos.org/t/buildgomodule-fails-due-to-invalid-go-version/40823)

File: pkgs/by-name/do/dotcopy/package.nix (hunk 3 above)
Comment (FireSquid6) [reply pkgs/by-name/do/dotcopy/package.nix:22]: it won't be in a few days so I figured I'd go ahead and set it up so I didn't have to waste someone's time with another PR

File: pkgs/by-name/do/dotcopy/package.nix (hunk 4 above)
Comment (FireSquid6) [reply pkgs/by-name/do/dotcopy/package.nix:23]: gpl3Only

**Comment by FireSquid6** [comment]
will fix all of this tomorrow. Thanks for the review!

File: pkgs/by-name/do/dotcopy/package.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/by-name/do/dotcopy/package.nix:22]:
```suggestion
    homepage = "https://dotcopy.firesquid.com/";
```

File: pkgs/by-name/do/dotcopy/package.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/by-name/do/dotcopy/package.nix:10]: Try again.

**Comment by FireSquid6** [comment]
alright I fucked shit up too much and don't know how to fix my git fuckups so I'm gonna close this. I'll make a new PR when dotcopy hits v1. Thanks for the learning experience.

**Pull request closed without merging**